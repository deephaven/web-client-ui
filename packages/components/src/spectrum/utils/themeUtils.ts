import { theme } from '@react-spectrum/theme-default';
import { themeSpectrumClassesCommon } from '../../theme/theme-spectrum';

export {
  useStyleProps,
  baseStyleProps,
  viewStyleProps,
} from '@react-spectrum/utils';

const { global, light, dark, medium, large } = theme;

/**
 * Extend light + dark theme variables with DH defaults.
 *
 * A theme is just a mapped collection of css class names that are generated
 * from a collection of css modules.
 *
 * e.g.
 * {
 *   global: {
 *     spectrum: 'spectrum_9e130c',
 *     'spectrum--medium': 'spectrum--medium_9e130c',
 *     'spectrum--large': 'spectrum--large_9e130c',
 *     'spectrum--darkest': 'spectrum--darkest_9e130c',
 *     'spectrum--dark': 'spectrum--dark_9e130c',
 *     'spectrum--light': 'spectrum--light_9e130c',
 *     'spectrum--lightest': 'spectrum--lightest_9e130c',
 *   },
 *   light: {
 *     'spectrum--light': 'spectrum--light_a40724',
 *     'dh-spectrum-theme--light': '_dh-spectrum-theme--light_1hblg_22',
 *   },
 *   dark: {
 *     'spectrum--darkest': 'spectrum--darkest_256eeb',
 *     'dh-spectrum-theme--dark': '_dh-spectrum-theme--dark_f7kge_22',
 *   },
 *   medium: {
 *     'spectrum--medium': 'spectrum--medium_4b172c',
 *   },
 *   large: {
 *     'spectrum--large': 'spectrum--large_c40598',
 *   },
 * }
 */
/* eslint-disable import/prefer-default-export */
export const themeDHDefault = {
  global,
  light: {
    ...light,
    ...themeSpectrumClassesCommon,
  },
  dark: {
    ...dark,
    ...themeSpectrumClassesCommon,
  },
  // scales
  medium,
  large,
};
