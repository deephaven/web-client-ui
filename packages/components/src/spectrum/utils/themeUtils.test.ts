import { themeDHDefault } from './themeUtils';

describe('themeDHDefault', () => {
  it('should merge Spectrum default with DH custom styles', () => {
    expect(themeDHDefault).toMatchSnapshot();
  });
});
