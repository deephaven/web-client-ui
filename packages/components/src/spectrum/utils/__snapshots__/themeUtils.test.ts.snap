// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themeDHDefault should merge Spectrum default with DH custom styles 1`] = `
{
  "dark": {
    "spectrum--darkest": "gP7gvq_spectrum--darkest",
    "theme-spectrum-alias": "theme-spectrum-alias",
    "theme-spectrum-palette": "theme-spectrum-palette",
  },
  "global": {
    "spectrum": "zA6MfG_spectrum",
    "spectrum--dark": "zA6MfG_spectrum--dark",
    "spectrum--darkest": "zA6MfG_spectrum--darkest",
    "spectrum--large": "zA6MfG_spectrum--large",
    "spectrum--light": "zA6MfG_spectrum--light",
    "spectrum--lightest": "zA6MfG_spectrum--lightest",
    "spectrum--medium": "zA6MfG_spectrum--medium",
  },
  "large": {
    "spectrum--large": "HAZavG_spectrum--large",
  },
  "light": {
    "spectrum--light": "PFjRbG_spectrum--light",
    "theme-spectrum-alias": "theme-spectrum-alias",
    "theme-spectrum-palette": "theme-spectrum-palette",
  },
  "medium": {
    "spectrum--medium": "xSyFOq_spectrum--medium",
  },
}
`;
